### PR TITLE
Move ReadTheDocs configuration to use YAML config

### PR DESCRIPTION
**Pull request by cgranade**

### CODE REVIEW

**Comment by cgranade** [comment]
This PR fixes build issues with the ReadTheDocs pipeline by moving over to a YAML-based configuration that gives us a bit more control. In particular, we can then move QuTiP to install after Cython has been installed, working around a bug in QuTiP's `setup.py`.

File: doc/environment.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  - pytest-cov
+  - pytest-xdist
+  - coveralls
+  - pylint
+  - pip:
+    # We also add additional tools for plotting.
+    - mpltools
+    # We use the PyPI version of QuTiP here since the conda version
+    # is on the wrong channel for use with RTD.
+    - qutip
```
Comment (ihincks) [inline doc/environment.yml]: Maybe you know, but you can set the channels at the top of the yml under  a ``channels:`` title. But I'm not familiar with this particular qutip quirk...no need to change it if it works

**Comment by ihincks** [comment]
Out of interest, could you briefly explain why qutip needs this special attention? (I may be adding it as a requirement to another project sometime soon)

File: doc/environment.yml (hunk 1 above)
Comment (cgranade) [reply doc/environment.yml]: I've used that in other places, but for some reason I was under the mistaken impression that ReadTheDocs didn't support `channels:`. I'll investigate that soon.

**Comment by cgranade** [comment]
The basic problem with including QuTiP is that its `setup.py` script needs Cython to already be installed. When pip tries to collect a list of dependencies it fails due to Cython not being present. The conda-forge recipe for QuTiP avoids this problem, as does installing with pip after Cython has been successfully installed.

File: doc/environment.yml (hunk 1 above)
Comment (ihincks) [reply doc/environment.yml]: Ooh, I have no idea what RTD accepts! I was just thinking about conda env.

**Pull request merged**